src/dynamic/LI_generator.cpp

```diff
@@ -79,7 +79,7 @@ std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> create_LI_model(uint
         std::vector<std::shared_ptr<cadmium::dynamic::modeling::model>> atomics_current_level;
         if (level < depth) {
             //Last level does not have atomics
-            for(int idx_atomic=0; idx_atomic < width; idx_atomic++) {
+            for(int idx_atomic=0; idx_atomic < width-1; idx_atomic++) {
```

Comment:
I would extract that "width -1" as  variable/constant and put a comment of why it is -1. We already missed-by-one error on it, lets leave a warning for future generations :)